capellambse_context_diagrams/collectors/custom.py

```diff
@@ -0,0 +1,462 @@
+        if self.repeat_depth > 0:
+            self.repeat_depth += len(targets)
+        for i in targets:
+            attr = i.get("name")
+            assert attr, "Attribute name is required."
+            target = getattr(obj, attr, None)
+            if isinstance(target, cabc.Iterable):
+                filters = i.get("filter", {})
+                for item in target:
+                    if item.uuid in self.visited:
+                        continue
+                    self.visited.add(item.uuid)
+                    if not self._matches_filters(item, filters):
+                        continue
+                    if create:
+                        self._make_target(item)
+                    self._perform_instructions(item, i)
+            elif isinstance(target, m.ModelElement):
+                if target.uuid in self.visited:
+                    continue
+                self.visited.add(target.uuid)
+                if create:
+                    self._make_target(target)
+                self._perform_instructions(target, i)
+
+    def _make_target(
+        self, obj: m.ModelElement
+    ) -> _elkjs.ELKInputChild | _elkjs.ELKInputEdge | None:
+        if _is_edge(obj):
+            return self._make_edge_and_ports(obj)
+        return self._make_box(obj, slim_width=self.diagram._slim_center_box)
+
+    def _make_box(
+        self,
+        obj: m.ModelElement,
+        **kwargs: t.Any,
+    ) -> _elkjs.ELKInputChild:
+        if box := self.boxes.get(obj.uuid):
+            return box
+        box = makers.make_box(
+            obj,
+            no_symbol=self.diagram._display_symbols_as_boxes,
+            **kwargs,
+        )
+        self.boxes[obj.uuid] = box
+        if self.diagram._display_unused_ports:
+            for attr in generic.DIAGRAM_TYPE_TO_CONNECTOR_NAMES[
+                self.diagram.type
+            ]:
+                for port in getattr(obj, attr, []):
+                    self._make_port_and_owner(port)
+        if self.diagram._display_parent_relation:
+            self.common_owners.add(
+                generic.make_owner_boxes(
+                    obj, self.diagram_target_owners, self._make_owner_box
+                )
+            )
+        return box
+
+    def _make_owner_box(
+        self,
+        obj: t.Any,
+    ) -> t.Any:
+        parent_box = self._make_box(
+            obj.owner,
+            layout_options=makers.DEFAULT_LABEL_LAYOUT_OPTIONS,
+        )
+        assert (obj_box := self.boxes.get(obj.uuid))
+        for box in (children := parent_box.children):
+            if box.id == obj.uuid:
+                break
+        else:
+            children.append(obj_box)
+            obj_box.width = max(
+                obj_box.width,
+                parent_box.width,
+            )
+            for label in parent_box.labels:
+                label.layoutOptions = makers.DEFAULT_LABEL_LAYOUT_OPTIONS
+        self.boxes_to_delete.add(obj.uuid)
+        return obj.owner
```

Comment:
Is almost the same as `_make_owner_box` from default.py. Except we modify the `width` of the box later on and variable names. Also the `_display_symbols_as_boxes` isn't used here. These inconsistencies probably stem from 2 separate implementations.
We should have only one generic function.